src/wallet/wallet.h

```diff
@@ -111,6 +116,13 @@ class CKeyPool
             READWRITE(nVersion);
         READWRITE(nTime);
         READWRITE(vchPubKey);
+        if (nVersion >= FEATURE_HD_SPLIT)
```

Comment:
I do not believe the serializer version is set to the wallet's version. I believe it is the one set in src/wallet/db.h - CLIENT_VERSION.